File: src/main/java/foundation/esoteric/minecraft/plugins/template/PaperTemplatePlugin.java
package foundation.esoteric.minecraft.plugins.template;

import dev.jorel.commandapi.CommandAPI;
import dev.jorel.commandapi.CommandAPIBukkitConfig;
import org.bukkit.plugin.java.JavaPlugin;

public final class PaperTemplatePlugin extends JavaPlugin {

  @Override
  public void onEnable() {
    getDataFolder().mkdir();
    saveDefaultConfig();

    CommandAPIBukkitConfig commandAPIConfig = new CommandAPIBukkitConfig(this);

    CommandAPI.onLoad(commandAPIConfig);
    CommandAPI.onEnable();
  }
}

File: src/main/java/foundation/esoteric/minecraft/plugins/template/PaperTemplatePlugin.kt
package foundation.esoteric.minecraft.plugins.template

import dev.jorel.commandapi.CommandAPI
import dev.jorel.commandapi.CommandAPIBukkitConfig
import org.bukkit.plugin.java.JavaPlugin

class PaperTemplatePlugin : JavaPlugin() {
    override fun onEnable() {
        dataFolder.mkdir()
        saveDefaultConfig()

        val commandAPIConfig = CommandAPIBukkitConfig(this)

        CommandAPI.onLoad(commandAPIConfig)
        CommandAPI.onEnable()
    }
}
